metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/aop/ReleaseDateAspect.java

```diff
@@ -72,13 +81,38 @@ public Object filterOnReleaseDateAdviceFindOne(ProceedingJoinPoint proceedingJoi
     }
 
     private Object getObjectUnlessBeforeReleaseDate(Object result) {
-        if (result instanceof Auditable) {
-            LocalDate releaseDate = ((Auditable) result).getReleaseDate();
-            if (releaseDate != null && releaseDate.isAfter(LocalDate.now())) {
-                return null;
-            }
+        if (!(result instanceof Auditable)) {
+            return result;
+        }
+
+        LocalDate releaseDate = ((Auditable) result).getReleaseDate();
+        if (releaseDate == null || !releaseDate.isAfter(LocalDate.now())) {
+            return result;
+        }
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication instanceof AnonymousAuthenticationToken) {
+            throw new SecurityException("Authentication required");
+        }
+
+        //Allow admins to access before release date
+        if (authentication.getAuthorities().stream()
+                .anyMatch(authority -> authority.getAuthority()
+                        .equals("ROLE_" + EnableSecurityConfig.ROLE_SERVICE_OPERATOR))) {
+            return result;
+        }
+
+        String commaSeperatedPermittedStudiesToTheUsers = ((CustomUsernamePasswordAuthenticationToken)
```

Comment:
it is kind of self-documenting variable name

Reply:
The method name is getObjectUnlessBeforeReleaseDate and documentation has " If the returned
 object" which are generic. If you want to make Study specific please mention it somewhere

Reply:
It would be sensible to rename to `commaSeparatedPermittedStudies` (since we're working in the context of a specific user anyway, and this is always implied). Also note the misspelling: _seperated_ → _separated_